yt/yt/server/job_proxy/user_job.cpp

```diff
@@ -661,58 +661,53 @@ class TUserJob
     void InitShellManager()
     {
 #ifdef _linux_
+        std::vector<TString> shellEnvironment;
+        shellEnvironment.reserve(Environment_.size());
+        std::vector<TString> visibleEnvironment;
+        visibleEnvironment.reserve(Environment_.size());
+
+        for (const auto& variable : Environment_) {
+            if (variable.StartsWith(NControllerAgent::SecureVaultEnvPrefix) && !UserJobSpec_.enable_secure_vault_variables_in_job_shell()) {
+                continue;
+            }
+            if (variable.StartsWith("YT_")) {
```

Comment:
Rebase it, please.

There should be conflict with concurrent change (ignore_yt_variables_in_shell_environment).